libraries/AP_HAL_PX4/RCOutput.cpp

```diff
@@ -495,8 +469,7 @@ void PX4RCOutput::_send_outputs(void)
                             ap_uc->rco_write(_period[i], i);
                         }
 
-                        bool armed = hal.util->get_soft_armed();
-                        if (armed) {
+                        if (hal.util->safety_switch_state() != AP_HAL::Util::SAFETY_DISARMED) {
```

Comment:
Well, `hal.util->get_soft_armed();` is actually used a lot in code to determine armed state (in very many places of vehicle specific code).
As for servos, in VRBrain RCOutput it is used also for same.
However I checked for `hal.util->safety_switch_state() != AP_HAL::Util::SAFETY_DISARMED` and it is also used a lot and actually by `update_soft_armed()` the `hal.util->get_soft_armed();` is reassembling the safety switch but also adds `arming.is_armed()` check.

So I believe correct will be: `arming.is_armed() && hal.util->safety_switch_state() != AP_HAL::Util::SAFETY_DISARMED` if we want to refuse from using `get_soft_armed();`